pkg/extensions/search/resolver.go

```diff
@@ -544,9 +545,120 @@ func repoListWithNewestImage(
+		return nil, err
+	}
+
+	starredRepos, err := repoDB.GetStarredRepos(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	for _, repoMeta := range reposMeta {
+		isStarred := zcommon.Contains(starredRepos, repoMeta.Name)
+		isBookmarked := zcommon.Contains(bookmarkedRepos, repoMeta.Name)
+
 		repoSummary := convert.RepoMeta2RepoSummary(ctx, repoMeta, manifestMetaMap, indexDataMap,
-			skip, cveInfo)
+			isStarred, isBookmarked, skip, cveInfo)
+		repos = append(repos, repoSummary)
+	}
+
+	paginatedRepos.Page = &gql_generated.PageInfo{
+		TotalCount: pageInfo.TotalCount,
+		ItemCount:  pageInfo.ItemCount,
+	}
+	paginatedRepos.Results = repos
+
+	return paginatedRepos, nil
+}
+
+func getBookmarkedRepos(
+	ctx context.Context,
+	cveInfo cveinfo.CveInfo,
+	log log.Logger, //nolint:unparam // may be used by devs for debugging
+	requestedPage *gql_generated.PageInput,
+	repoDB repodb.RepoDB,
+) (*gql_generated.PaginatedReposResult, error) {
+	repoNames, err := repoDB.GetBookmarkedRepos(ctx)
+	if err != nil {
+		return &gql_generated.PaginatedReposResult{}, err
+	}
+
+	filterFn := func(repoMeta repodb.RepoMetadata) bool {
+		return zcommon.Contains(repoNames, repoMeta.Name)
+	}
+
+	return getFilteredPaginatedRepos(ctx, cveInfo, filterFn, log, requestedPage, repoDB)
+}
+
+func getStarredRepos(
+	ctx context.Context,
+	cveInfo cveinfo.CveInfo,
+	log log.Logger, //nolint:unparam // may be used by devs for debugging
+	requestedPage *gql_generated.PageInput,
+	repoDB repodb.RepoDB,
+) (*gql_generated.PaginatedReposResult, error) {
+	repoNames, err := repoDB.GetStarredRepos(ctx)
+	if err != nil {
+		return &gql_generated.PaginatedReposResult{}, err
+	}
+
+	filterFn := func(repoMeta repodb.RepoMetadata) bool {
+		return zcommon.Contains(repoNames, repoMeta.Name)
+	}
+
+	return getFilteredPaginatedRepos(ctx, cveInfo, filterFn, log, requestedPage, repoDB)
+}
+
+func getFilteredPaginatedRepos(
+	ctx context.Context,
+	cveInfo cveinfo.CveInfo,
+	filterFn repodb.FilterRepoFunc,
+	log log.Logger, //nolint:unparam // may be used by devs for debugging
+	requestedPage *gql_generated.PageInput,
+	repoDB repodb.RepoDB,
+) (*gql_generated.PaginatedReposResult, error) {
+	repos := []*gql_generated.RepoSummary{}
+	paginatedRepos := &gql_generated.PaginatedReposResult{}
+
+	if requestedPage == nil {
+		requestedPage = &gql_generated.PageInput{}
+	}
+
+	skip := convert.SkipQGLField{
+		Vulnerabilities: canSkipField(convert.GetPreloads(ctx), "Results.NewestImage.Vulnerabilities"),
+	}
+
+	pageInput := repodb.PageInput{
+		Limit:  safeDerefferencing(requestedPage.Limit, 0),
+		Offset: safeDerefferencing(requestedPage.Offset, 0),
+		SortBy: repodb.SortCriteria(
+			safeDerefferencing(requestedPage.SortBy, gql_generated.SortCriteriaUpdateTime),
+		),
+	}
+
+	reposMeta, manifestMetaMap, indexDataMap, pageInfo, err := repoDB.FilterRepos(ctx, filterFn, pageInput)
+	if err != nil {
+		return paginatedRepos, err
+	}
+
+	bookmarkedRepos, err := repoDB.GetBookmarkedRepos(ctx)
```

Comment:
This logic is duplicated. FilterRepos in this case also accepts a function that does the same thing

Reply:
Unless we return that information as part of the repo metadata directly from repodb, this will have to do.